frontend/appflowy_tauri/src/appflowy_app/components/document/CodeBlock/index.tsx

```diff
@@ -1,3 +1,39 @@
-export default function CodeBlock({ id }: { id: string }) {
-  return <div>{id}</div>;
+import { BlockType, NestedBlock } from '$app/interfaces/document';
+import { useCodeBlock } from './CodeBlock.hooks';
+import { Editable, Slate } from 'slate-react';
+import BlockHorizontalToolbar from '$app/components/document/BlockHorizontalToolbar';
+import React from 'react';
+import { CodeLeaf, CodeBlockElement } from './elements';
+import SelectLanguage from './SelectLanguage';
+import { decorateCodeFunc } from '$app/utils/document/blocks/code/decorate';
+
+export default function CodeBlock({
+  node,
+  placeholder,
+  ...props
+}: { node: NestedBlock<BlockType.CodeBlock>; placeholder?: string } & React.HTMLAttributes<HTMLDivElement>) {
+  const { editor, value, onChange, onKeyDown, onDOMBeforeInput } = useCodeBlock(node);
+
+  const className = props.className !== undefined ? ` ${props.className}` : '';
```

Comment:
a quick question. Is the `x !== undefined ? a : b` equal to `x ? a : b`?

Reply:
Almost, null also exists in javascript, so in this case if the value was set to null somewhere, then it wouldn't be `undefined`.

```suggestion
  const className = props.className ? ` ${props.className}` : '';
```
I think this is safer.

Reply:
I would have expected all values ​​except undefined to be displayed directly, including 0 and null

Reply:
If `props.className` for some reason would be null here, then `className` would be `' null'`.